src/design-system/components/badge/Badge.tsx

```diff
@@ -0,0 +1,42 @@
+import { Badge as MuiBadge, BadgeProps as MuiBadgeProps } from '@mui/material'
+import { FC } from 'react'
+
+type BadgeColor = 'primary' | 'success' | 'error'
+
+type SmallBadgeProps = {
+  variant: 'sm'
+  color?: BadgeColor
+  isVisible?: boolean
+}
+
+type LargeBadgeProps = {
+  variant: 'lg'
+  badgeContent: number
+  maxContent?: number
+  color?: BadgeColor
+  isVisible?: boolean
+  showZero?: boolean
+}
+
+type BadgeProps = (LargeBadgeProps | SmallBadgeProps) &
+  Omit<MuiBadgeProps, 'variant'>
+
+const Badge: FC<BadgeProps> = ({ children, ...props }) => {
```

Comment:
Do not import FC from react, just use React.FC instead. It it more clean and readable
```suggestion
const Badge: React.FC<BadgeProps> = ({ children, ...props }) => {
```